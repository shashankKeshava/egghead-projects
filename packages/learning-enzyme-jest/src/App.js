import React, {Component} from "react";
import logo from "./logo.svg";
import "./App.css";

const Test = ({text}) => <div>{text}</div>;
class App extends Component {
    componentDidMount() {
        this.setState({lifeCycle: "componentDidMount"});
    }
    componentWillReceiveProps() {
        this.setState({lifeCycle: "componentWillReceiveProps"});
    }
    render() {
        return (
            <div className="App">
                <header className="App-header">
                    <img src={logo} className="App-logo" alt="logo" />
                    <p className="App-intro">
                        Edit <code>src/App.js</code> and save to reload.
                    </p>
                    <a
                        className="App-link"
                        href="https://reactjs.org"
                        target="_blank"
                        rel="noopener noreferrer"
                    >
                        Learn React
                    </a>
                </header>
                <Test text={"Testing Components with Props"} />
                <ul>
                    <li>List 1</li>
                    <li>List 2</li>
                    <li>List 3</li>
                </ul>
            </div>
        );
    }
}

class Link extends Component {
    render() {
        return <a href={this.props.address}>Click</a>;
    }
}

export {Link, App};
