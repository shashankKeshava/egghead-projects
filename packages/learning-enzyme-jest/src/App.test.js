import React from "react";
import ReactDOM from "react-dom";
import {App, Link} from "./App";
import {configure, shallow, mount} from "enzyme";
import Adapter from "enzyme-adapter-react-16.3";
import toJson from "enzyme-to-json";

configure({adapter: new Adapter()});

describe("<App/> shallow rendering", () => {
    const wrapper = shallow(<App />);
    it("should contain 1 p element", () => {
        expect(wrapper.find(".App-intro").exists()).toBe(true);
    });
    it("should contain 3 li element", () => {
        expect(wrapper.find("ul").children().length).toBe(3);
    });
    it("should match snapshot", () => {
        const tree = shallow(<App />);
        expect(toJson(tree)).toMatchSnapshot();
    });
});

describe("<App/> mount rendering", () => {
    it("should contain 1 p element", () => {
        const wrapper = mount(<App />);
        expect(wrapper.find(".App-intro").exists()).toBe(true);
        wrapper.unmount();
    });
    it("should contain 3 li element", () => {
        const wrapper = mount(<App />);
        expect(wrapper.find("ul").children().length).toBe(3);
        wrapper.unmount();
    });
    it("should match snapshot", () => {
        const tree = mount(<App />);
        expect(toJson(tree)).toMatchSnapshot();
        tree.unmount();
    });
});

describe("<Link/>", () => {
    it("link component accepts address props", () => {
        const wrapper = shallow(<Link address="www.google.com" />);
        expect(wrapper.instance().props.address).toBe("www.google.com");
    });
    it("a tag renders href correctly", () => {
        const wrapper = shallow(<Link address="www.facebook.com" />);
        expect(wrapper.props().href).toBe("www.facebook.com");
    });
});
