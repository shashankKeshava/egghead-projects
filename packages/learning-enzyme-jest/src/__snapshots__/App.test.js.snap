// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> shallow rendering should match snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <p
      className="App-intro"
    >
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
  <Test
    text="Testing Components with Props"
  />
  <ul>
    <li>
      List 1
    </li>
    <li>
      List 2
    </li>
    <li>
      List 3
    </li>
  </ul>
</div>
`;
